c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+#include "Steam.h"
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenMainMenu()
+{
+	system("cls");
+
+	const std::string MainMenuMessage = "Welcome to the Steam store! What would you like to do? \n1 - Add a game\n2 - Modify or create game categories \n3 - View all games \n4 - Exit \nTELL ME: ";
+	std::cout << MainMenuMessage;
+
+	enum class EMainMenuCommand { AddGameMenu = 1, CategoryManagerMenu, GameDisplayerMenu, Exit };
+	EMainMenuCommand UserChoice;
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	UserChoice = (EMainMenuCommand)ValidateInputAndReturnValidatedInput(TemporalUserChoice, MainMenuMessage, 1, 4);
+
+	switch (UserChoice)
+	{
+	case EMainMenuCommand::AddGameMenu:
+		ActiveCommand = EMenuCommand::AddGamesMenu;
+		break;
+
+	case EMainMenuCommand::CategoryManagerMenu:
+		ActiveCommand = EMenuCommand::CategoryManagerMenu;
+		break;
+
+	case EMainMenuCommand::GameDisplayerMenu:
+		ActiveCommand = EMenuCommand::GameDisplayerMenu;
+		break;
+
+	case EMainMenuCommand::Exit:
+		ActiveCommand = EMenuCommand::Exit;
+		break;
+
+	default:
+		std::cout << "MainMenu Error, please contact support";
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenAddGamesMenu()
+{
+	system("cls");
+
+	// CategoryIndex is set to -1 and that value is representing the uncategorized category
+	int CategoryIndex = -1;
+
+	if (!CategoryContainer.IsEmpty())
+	{
+		CategoryIndex = ChooseCategory();
```

Comment:
This is a bit overcomplicated and dangerous (in the sense that it is not accounting for user error). Your `ChooseCategory` function is always assuming that you only call it if the category container is empty... but couldn't that be part of the ChooseCategory function itself?

Instead of 
```
if (!AskChooseCategory())
{
    return -1;
}
```
You could have
```
if (CategoryContainer.IsEmpty() || !AskChooseCategory())
{
    return -1;
}
```
and this way, you don't have to concern yourself with validating the CategoryContainer before calling ChooseCategory() and ChooseCategory() wouldn't be possibly used wrong and cause a weird bug.